pkg/util/provider/machinecontroller/machine.go

```diff
@@ -325,12 +325,12 @@ func (c *controller) triggerCreationFlow(ctx context.Context, createMachineReque
 			Secret:       createMachineRequest.Secret,
 		},
 	)
-	if err == nil {
+	if err == nil && (nodeName == "" || providerID == "") {
```

Comment:
I did not understand how this condition is making any difference with the previous logic as `(nodeName == "" || providerID == "")` is always `True` looking at the code flow.

Reply:
I used the logic that once it enters this `if` then `nodeName` and `providerID` are sure to be initialized. so when the flow will not enter this `if` the next time because of the condition `(nodeName == "" || providerID == "")`

Reply:
But in L307 and L308, these variables are initialised with empty strings. So, I am not understanding how the subsequent reconciliation calls will behave any different.

Reply:
hmm. right , I should have used 
`nodeName` -> `machine.Labels["node"]`
`providerID` -> `machine.Spec.providerID`

instead. Good catch , I hope making these changes if fine ?

Reply:
I have addressed the changes here 
68a9577
It works as expected and have run the integration test locally as well.